packages/headless/src/api/analytics/base-analytics.ts

```diff
@@ -23,19 +21,13 @@ export type StateNeededByBaseAnalyticsProvider = ConfigurationSection &
 
 export abstract class BaseAnalyticsProvider<
   T extends StateNeededByBaseAnalyticsProvider,
-> implements SearchPageClientProvider
-{
+> {
   protected readonly state: T;
   constructor(protected getState: () => T) {
     this.state = getState();
   }
 
   public abstract getSearchUID(): string;
-  public abstract getPipeline(): string;
-  public abstract getSearchEventRequestPayload(): Omit<
-    SearchEventRequest,
-    'actionCause' | 'searchQueryUid'
-  >;
```

Comment:
@coveo/dx was anyone relying on these abstract to be there in the BaseAnalyticsProvider?

Are they safe to remove.

I'm assuming the only thing it means is that any AnalyticsProvider now needs to know to implement those if they want to be a searchAnalyticsProvider?

I'm assuming @mmitiche you'd already validated, but I checked too and no code referencing a `BaseAnalyticsProvider` as a type was calling `getPipeline` or `getSearchEventRequestPayload` for example. 

Probably only more specific types were calling those methods.

Reply:
If the test passes, we should be good normally.